src/main/java/org/pkcs11/jacknji11/GetAttributeProcess.java

```diff
@@ -0,0 +1,381 @@
+package org.pkcs11.jacknji11;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.pkcs11.jacknji11.CK.UNAVAILABLE_INFORMATION;
+import static org.pkcs11.jacknji11.CKA.MODULUS;
+import static org.pkcs11.jacknji11.CKA.allocate;
+import static org.pkcs11.jacknji11.CKA.indefinite;
+
+/**
+ * Encapsulates attribute fetching process.
+ * <p>
+ * The process iteratively fetches all attributes.
+ * It operates either in batch mode or one-by-one.
+ * <p>
+ * The process is able to use max length of attribute instead of querying for attribute lengths,
+ * which can save some time by not issuing additional queries.
+ * <p>
+ * At worst case it will issue 3 queries to fetch a requested attribute, at best only 1.
+ * It is optimized for fetching many attributes at once (batch mode).
+ * <p>
+ * Default sizes for large and small attributes are:
+ * <ul>
+ *     <li>DEFAULT_LARGE_ATTRIBUTE_LENGTH = 2048</li>
+ *     <li>DEFAULT_ATTRIBUTE_LENGTH = 72</li>
+ * </ul>
+ *
+ * One can change those defaults by setting system properties:
+ * <ul>
+ *     <li>org.pkcs11.jacknji11.GetAttributeProcess.DEFAULT_LARGE_ATTRIBUTE_LENGTH</li>
```

Comment:
Can we have these settable by code as well. System properties are not ideal when using the library from a larger application, for example running in an application server, inside a container.

Reply:
Oh, and document above (and in setters) as well what setting them to 0 mean.

Reply:
If that is your wish then I will make those configurable only via code.  As I see it the configuration is per provider therefore I will move those config options to CryptokiE class.